app/assets/javascripts/select-kit/components/topic-notifications-button.js.es6

```diff
@@ -1,15 +1,24 @@
 import Component from "@ember/component";
+import { computed } from "@ember/object";
+import { later } from "@ember/runloop";
 
 export default Component.extend({
   layoutName: "select-kit/templates/components/topic-notifications-button",
   classNames: ["topic-notifications-button"],
   appendReason: true,
   showFullTitle: true,
+  isLoading: false,
+  icon: computed("isLoading", function() {
+    return this.isLoading ? "spinner" : null;
+  }),
 
   actions: {
     changeTopicNotificationLevel(newNotificationLevel) {
       if (newNotificationLevel !== this.notificationLevel) {
-        this.topic.details.updateNotifications(newNotificationLevel);
+        this.set("isLoading", true);
+        this.topic.details
+          .updateNotifications(newNotificationLevel)
+          .finally(() => later(() => this.set("isLoading", false), 250));
```

Comment:
Why do we need the 250ms delay here?

Reply:
Sorry forgot to answer, this is to have a consistent loading animation. Often the loading is so fast that you barely see it and end up just having a blinking/clunky UI. Given this is only for animation purpose and doesn't block UI I think it's fine to always have this delay.

Reply:
Sounds good to me. Thanks for the clarification.